packages/stryker/src/MutatorOrchestrator.ts

```diff
@@ -130,6 +130,27 @@ export default class MutatorOrchestrator {
       }
     });
 
-    return mutants;
+    return this.reduceMutantSize(mutants, this.options.proportionOfMutantsToUse);
+  }
+
+  private reduceMutantSize(mutants: Mutant[], proportionToKeep: number) {
+    const mutantMap = {};
+    mutants.forEach(mutant => {
+      const mutantsOfThisType = mutantMap[mutant.mutatorName];
+      if (!mutantsOfThisType) {
+        mutantMap[mutant.mutatorName] = [mutant];
+      } else {
+        mutantsOfThisType.push(mutant);
+      }
+    });
+
+    let mutantsThatMakeTheCut: Mutant[] = [];
+    Object.keys(mutantMap).forEach(mutantType => {
+      const mutantsOfThisType: Mutant[] = mutantMap[mutantType];
+      const numberOfThisTypeToKeep = Math.ceil(proportionToKeep * mutantsOfThisType.length);
+      mutantsThatMakeTheCut = mutantsThatMakeTheCut.concat(mutantsOfThisType.slice(0, numberOfThisTypeToKeep));
```

Comment:
Isn't it better to take every *nth* mutant instead of slicing off the first n mutants? That way, the mutants will be more evenly spread among the source files.